app/screens/image_preview/downloader.ios.js

```diff
@@ -76,82 +97,149 @@ export default class Downloader extends PureComponent {
+                    progress: 100
+                }, () => {
+                    // need to wait a bit for the progress circle UI to update to the give progress
+                    setTimeout(async () => {
+                        try {
+                            // handles the case of a late cancellation by the user
+                            // and ensures that we remove the file if they did cancel late.
+                            if (this.state.didCancel) {
+                                // TODO: There's issue with deleting files from the cameraRoll on iOS here https://github.com/wkh237/react-native-fetch-blob/issues/479
+                                // This only occurs if the user cancels when the download is around 80% or more
+                                await RNFetchBlob.fs.unlink(path);
+                                this.props.onDownloadCancel();
+                            } else {
+                                this.props.onDownloadSuccess();
+                            }
+                        } catch (error) {
+                            // ensure the downloader at least closes if a error
                             this.props.onDownloadCancel();
-                        } else {
-                            this.props.onDownloadSuccess();
                         }
-                    } catch (error) {
-                        // ensure the downloader at least closes if a error
-                        this.props.onDownloadCancel();
-                    }
-                }, 2000);
-            });
+                    }, 2000);
+                });
+            }
 
-            res.flush(); // remove the temp file
+            if (saveToCameraRoll) {
+                res.flush(); // remove the temp file
+            }
             this.downloadTask = null;
         } catch (error) {
             // cancellation throws so we need to catch
+            if (downloadPath) {
+                RNFetchBlob.fs.unlink(`${downloadPath}/${file.id}.${file.extension}`);
+            }
+            if (error.message !== 'cancelled' && this.mounted) {
+                const {intl} = this.context;
+                Alert.alert(
+                    intl.formatMessage({
+                        id: 'mobile.downloader.failed_title',
+                        defaultMessage: 'Download failed'
+                    }),
+                    intl.formatMessage({
+                        id: 'mobile.downloader.failed_description',
+                        defaultMessage: 'An error occurred while downloading the file. Please check your internet connection and try again.\n'
+                    }),
+                    [{
+                        text: intl.formatMessage({
+                            id: 'mobile.server_upgrade.button',
+                            defaultMessage: 'OK'
+                        }),
+                        onPress: () => this.setState({progress: 0, started: false, didCancel: false})
```

Comment:
Maybe make this into a method because we're doing the same thing in multiple places?